### PR TITLE
directory structure organization

### PR DESCRIPTION
**Pull request by drohnow**
Repo directory structure reorg as per [shortcut 82412](https://app.shortcut.com/replicated/story/82412/setup-platform-examples-and-helm-starter-repo) for discussion and consideration purposes.

### CODE REVIEW

File: .github/workflows/lint-test.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Lint and Test Charts
```
Comment (diamonwiggins) [inline .github/workflows/lint-test.yaml:1]: Should this and `main.yaml` be at the root of the repo?

File: .helmignore [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Patterns to ignore when building packages.
```
Comment (diamonwiggins) [inline .helmignore:1]: same feedback as the github workflows at the root of the repo, i'm not sure they are meant to be here. Seems like you need to copy in the directory structure from https://github.com/replicatedhq/platform-examples/pull/1

**Review by diamonwiggins — CHANGES_REQUESTED** [review]
A few comments about some files that have been copied in. So far it looks like this is just a direct copy of the `examples` repo from the starter repo. I imagine some things can be removed and the initial scaffolding done in https://github.com/replicatedhq/platform-examples/pull/1 doesn't seem to be here. 

Also it may be helpful to add an example for the first pattern, so we can have an understanding of what that should look like on future PRs? What's documented in `applications/determined-ai/README.md` I think would qualify as patterns.

**Comment by drohnow** [comment]
> A few comments about some files that have been copied in. So far it looks like this is just a direct copy of the `examples` repo from the starter repo. I imagine some things can be removed and the initial scaffolding done in #1 doesn't seem to be here.
> 
> Also it may be helpful to add an example for the first pattern, so we can have an understanding of what that should look like on future PRs? What's documented in `applications/determined-ai/README.md` I think would qualify as patterns.

Is that a pattern or an application?  My understanding is that a pattern is a snippet or subset of a solution that can be reused.

I believe the intention here for this PR is to propose a directory so we can iterate on it.

File: .helmignore (hunk 2 above)
Comment (drohnow) [reply .helmignore]: agreed, removed.

File: patterns/.github/workflows/lint-test.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+name: Lint and Test Charts
```
Comment (diamonwiggins) [inline patterns/.github/workflows/lint-test.yaml:1]: Last thing. I don't think you'll want these here as the actions should be in each individual pattern sub dir

**Review by diamonwiggins — APPROVED** [review]
Tentative approval since there's one more dir that needs to be cleaned up.

**Comment by diamonwiggins** [comment]
> Is that a pattern or an application? My understanding is that a pattern is a snippet or subset of a solution that can be reused.

If i'm understanding correctly I think that each individual example i documented in the determined README would be a pattern use case.

File: patterns/.github/workflows/lint-test.yaml (hunk 3 above)
Comment (drohnow) [reply patterns/.github/workflows/lint-test.yaml:1]: > Last thing. I don't think you'll want these here as the actions should be in each individual pattern sub dir

@Chris Sanders I know we discussed yesterday and went with .github/workflows; but I think I like @diamonwiggins idea. What are your thoughts here?

Comment (chris-sanders) [reply patterns/.github/workflows/lint-test.yaml:1]: What's the suggestion here, `patterns/.github/workflows/lint/lint-test.yaml`?

Comment (diamonwiggins) [reply patterns/.github/workflows/lint-test.yaml:1]: To clarify, i'm not saying that the structure `.github/workflows` should change. I'm saying that inside of the `patterns` directory there will be sub directories for each pattern I imagine and that's where the workflows should live. Not at the root of `patterns/`

Comment (chris-sanders) [reply patterns/.github/workflows/lint-test.yaml:1]: after slack discussion `patterns/github-workflows/.github/wofklow/<all the yamls>` is the pattern we'll go with for now

**Comment by chris-sanders** [comment]
> If i'm understanding correctly I think that each individual example i documented in the determined README would be a pattern use case.

@diamonwiggins yea it could be, no hard rule that it has to be. The fact that they're all documented in the examples seems like a reasonable way to handle it as well. I honestly think we'll make the decision by which direction you come at it from. If you write an entire app, put the notes in it's readme. If you just have one pattern to document and it's not reasonable to write a while app just put it in patterns.

File: patterns/.github/workflows/lint-test.yaml (hunk 3 above)
Comment (diamonwiggins) [reply patterns/.github/workflows/lint-test.yaml:1]: 👍

**Comment by diamonwiggins** [comment]
@drohnow PR only needs to slight directory structure modification highlighting the github workflow pattern. After that, feel free to merge .

**Pull request merged**